### ISSUE TITLE
User labels are overwritten during runtime label sync

### ISSUE THREAD

**Issue by raybejjani** [cilium/cilium#4021]
When the runtime drivers periodically synchronises container labels, it overwrites the user provided overrides via `/v1/endpoint/{id}/labels POST`. This is a bug.
Relatedly, `/v1/endpoint/{id} POST` ignores the labels when set. This is also bug (`/v1/endpoint/{id}/labels` is a convenience endpoint, not the exclusive way to set the labels).

**Comment by ianvernon** [comment]
This might be fixed by the changes @rlenglet is working on for #3911 , as those changes affect how labels are plumbed / created for endpoints.

**Comment by raybejjani** [comment]
True, but it was @rlenglet that mentioned it in #development so it might have made his work more difficult instead?

**Comment by rlenglet** [comment]
@raybejjani yeah, I had to fix setting labels on endpoint creation, which was not working.

And in the process I realized that:
* setting labels in `PATCH /endpoint/{id}` is not working (we are just ignoring the labels that are passed)
* setting labels in `PATCH /endpoint/{id}/labels` is currently useless, because those are immediately overridden by the labels sync

If we fixed `PATCH /endpoint/{id}` it would be useless just like `PATCH /endpoint/{id}/labels` because of the labels sync.
I don't think there's an easy way to fix that.
